### PR TITLE
Use MediaQueryListEvent for media query list change events

### LINKED ISSUE #14064: Use MediaQueryListEvent for media query list change events
https://github.com/servo/servo/pull/13453 implemented the window.matchMedia and MediaQueryList APIs. However, the [specification](https://drafts.csswg.org/cssom-view/#evaluate-media-queries-and-report-changes) related to the "change" event was [unclear](https://github.com/w3c/csswg-drafts/issues/681), so we didn't notice that we should use a specific event interface instead of Event.

* [create](http://doc.servo.org/script/dom/index.html#adding-a-new-dom-interface) the missing [MediaQueryListEvent](https://drafts.csswg.org/cssom-view/#dom-mediaquerylistevent-mediaquerylistevent) interface (use existing event types like `PageTransitionEvent` as a model)
* modify `WeakMediaQueryListVec::evaluate_and_report_changes` to use the new event type and set its properties according to [the specification](https://drafts.csswg.org/cssom-view/#evaluate-media-queries-and-report-changes)

Code: `components/script/dom/webidls/MediaQueryListEvent.webidl`, `components/script/dom/mod.rs`, `components/script/dom/mediaquerylistevent.rs`, `components/script/dom/mediaquerylist.rs`

Tests:
* modify `tests/wpt/mozilla/tests/css/matchMedia.html` to assert that the change event received is `instanceof MediaQueryListEvent` (ensure `./mach test-wpt tests/wpt/mozilla/tests/css/matchMedia.html` passes)
* add the new interface to the list in `tests/wpt/mozilla/tests/mozilla/interfaces.html` (make sure `./mach test-wpt tests/wpt/mozilla/tests/mozilla/interfaces.html` passes)

### PR DESCRIPTION
**Pull request by bd339**
- Implements MediaQueryListEvent interface
- Makes MediaQueryList change events conform to the [spec](https://drafts.csswg.org/cssom-view/#evaluate-media-queries-and-report-changes)
- Updates matchMedia.html test to verify that the new event type is used

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #14064

- [X] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14630)

### CODE REVIEW

File: components/script/dom/mediaquerylist.rs [hunk 1]
```diff
@@ -137,7 +140,14 @@ impl WeakMediaQueryListVec {
     pub fn evaluate_and_report_changes(&self) {
         for mql in self.cell.borrow().iter() {
             if let MediaQueryListMatchState::Changed(_) = mql.root().unwrap().evaluate_changes() {
-                mql.root().unwrap().upcast::<EventTarget>().fire_event(atom!("change"));
+                let ev = MediaQueryListEvent::new(&mql.root().unwrap().global(),
```
Comment (KiChjang) [inline components/script/dom/mediaquerylist.rs]: I believe having a `let mql = mql.root().unwrap();` above this line would save you a lot of the unnecessary rooting and unwrapping every time you want to make a reference to the `MediaQueryList`.

File: tests/wpt/mozilla/tests/mozilla/interfaces.html [hunk 2]
```diff
@@ -143,6 +143,7 @@
   "MediaError",
   "MediaList",
   "MediaQueryList",
+  "MediaQueryListEvent",
```
Comment (KiChjang) [inline tests/wpt/mozilla/tests/mozilla/interfaces.html]: This needs review from @jdm or @Ms2ger in order to be added.

Comment (jdm) [inline tests/wpt/mozilla/tests/mozilla/interfaces.html]: I approve this change.

File: components/script/dom/mediaquerylist.rs [hunk 3]
```diff
@@ -137,7 +140,14 @@ impl WeakMediaQueryListVec {
...
+                                                  atom!("change"),
+                                                  false, false,
+                                                  mql.root().unwrap().Media(),
+                                                  mql.root().unwrap().Matches());
+                let event = ev.upcast::<Event>();
+                event.set_trusted(true);
+                mql.root().unwrap().upcast::<EventTarget>().dispatch_event(&event);
```
Comment (jdm) [inline components/script/dom/mediaquerylist.rs]: We should this instead:
```rust
let event = ev.upcast::<Event>();
event.fire(mql.root().unwrap().upcast::<EventTarget>());
```

Comment (bd339) [inline components/script/dom/mediaquerylist.rs]: Will do, but what is the difference?

Comment (KiChjang) [inline components/script/dom/mediaquerylist.rs]: `Event::fire` implicitly sets the trusted field to true.

Comment (bd339) [inline components/script/dom/mediaquerylist.rs]: Oh, right. Completely missed that from his snippet.

File: components/script/dom/mediaquerylistevent.rs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    // https://drafts.csswg.org/cssom-view/#dom-mediaquerylistevent-matches
+    fn Matches(&self) -> bool {
+        self.matches.get()
+    }
+
+    // https://dom.spec.whatwg.org/#dom-event-istrusted
+    fn IsTrusted(&self) -> bool {
+        self.event.IsTrusted()
```
Comment (KiChjang) [inline components/script/dom/mediaquerylistevent.rs]: `self.upcast::<Event>().IsTrusted()`

File: components/script/dom/mediaquerylistevent.rs [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+                                             DOMString::new(),
+                                             false)
+    }
+
+    pub fn new_initialized(global: &GlobalScope,
+                           media: DOMString,
+                           matches: bool) -> Root<MediaQueryListEvent> {
+        let ev = box MediaQueryListEvent {
+            event: Event::new_inherited(),
+            media: media.clone(),
```
Comment (KiChjang) [inline components/script/dom/mediaquerylistevent.rs]: Is the clone necessary here?

**Review by KiChjang — CHANGES_REQUESTED** [review]
Great work! One minor nit and you can r=me.

**Comment by bd339** [comment]
r=KiChjang

**Review by KiChjang — APPROVED** [review]

**Pull request merged**